ipam/client/backends/phpipam.py

```diff
@@ -258,6 +258,38 @@ def delete_ip(self, ipaddress):
         self.db.commit()
         return True
 
+    def delete_subnet(self, subnet, empty_subnet=False):
+        """Delete a subnet in IPAM. subnet must be an
+        instance of ip_network with correct prefix length.
+        If empty_subnet is True, we will remove all IP addresses
+        in the subnet. Otherwise, we will raise an exception.
+        """
+        subnet_id = self.find_subnet_id(subnet)
+        if subnet_id is None:
+            raise ValueError("Unable to get subnet id from database"
+                             "for subnet %s/%s"
+                             % (subnet.network_address,
+                                subnet.prefixlen))
+        self.cur.execute("SELECT ip_addr FROM ipaddresses \
+                         WHERE subnetId=%d"
+                         % (subnet_id))
+        row = self.cur.fetchone()
+        if row is not None:
+            # We have IP addresses in our subnet
+            if empty_subnet:
+                self.cur.execute("DELETE FROM ipaddresses \
+                                 WHERE subnetId=%d"
+                                 % subnet_id)
+            else:
+                raise ValueError("Subnet %s/%s is not empty"
+                                 % (subnet.network_address,
+                                    subnet.prefixlen))
```

Comment:
You have to check if this subnet has no child subnets in order to prevent creating orphan subnets.
You could leverage _get_allocated_subnets for that, and trigger an exception if this subnet has any children (this made me see that prefixlen is not used in this method, will fix this).

Reply:
FYI, I've just removed prefixlen parameter from _get_allocated_subnets